Обращение к /issues без параметра (Bug/37)

astronautr opened the pull request:
Добавил возможность отправляя GET-запрос по адресу localhost:8080/issues получать список всех `Issue`, находящихся в `issueMap`.

astronautr committed:
Fix a bug. Now you can send GET-request to /issues and retrieve list of issues

Card: #37

astronautr committed:
Small code reformat by IDE

Card: #37

astronautr commented:
![](https://github.trello.services/images/mini-trello-icon.png) [В IssueController (getIssuesByStatus) учесть ситуацию когда параметр запроса не подаётся.](https://trello.com/c/n8syjCDc/37-issuecontroller-getissuesbystatus)

mgtriffid commented on backend/tracker-web-app.backend/src/main/java/edu/xored/tracker/IssueController.java:
```diff
@@ -60,15 +57,20 @@ public Issue getIssue(@PathVariable("hash") long hash) {
+            return issueList;
+        }
+        issueList = new ArrayList<Issue>();
+        for (Map.Entry<Long, Issue> entry : issueMap.entrySet()) {
```
У нас есть Java 8, давай через Stream API? Ну, там, filter, collect, лямбды, вот это всё

astronautr replied:
Done.

astronautr committed:
Add usage of lambdas and filters for replacing filling loop

Card: #37

kevromster commented on backend/tracker-web-app.backend/src/main/java/edu/xored/tracker/IssueController.java:
```diff
@@ -3,10 +3,7 @@
-import java.util.Map;
-import java.util.List;
-import java.util.ArrayList;
+import java.util.*;
```
писал уже, звёздочки - bad style

astronautr replied:
Done.

kevromster commented on backend/tracker-web-app.backend/src/main/java/edu/xored/tracker/IssueController.java:
```diff
@@ -60,15 +57,19 @@ public Issue getIssue(@PathVariable("hash") long hash) {
+    public List<Issue> getIssues(@RequestParam(value = "status", required = false) Issue.Status status) {
+        List<Issue> issueList;
+        if (status == null) {
+            issueList = new ArrayList<Issue>(issueMap.values());
```
А вот вопрос, зачем мы _List_ возвращаем тут? Никакого порядка мы не гарантируем.
Тогда тут просто можно было бы _issueMap.values()_ вернуть

astronautr replied:
Done.

kevromster commented on backend/tracker-web-app.backend/src/main/java/edu/xored/tracker/IssueController.java:
```diff
@@ -60,15 +57,19 @@ public Issue getIssue(@PathVariable("hash") long hash) {
         }
-        return statusIssueList;
+        issueList = new ArrayList<Issue>();
+        issueMap.entrySet()
```
можно сразу фильтровать issueMap.values(), зачем нам entry set'ы

astronautr replied:
Done.

kevromster commented on backend/tracker-web-app.backend/src/main/java/edu/xored/tracker/IssueController.java:
```diff
@@ -60,15 +57,19 @@ public Issue getIssue(@PathVariable("hash") long hash) {
+        issueMap.entrySet()
+                .stream()
+                .filter(entry -> entry.getValue().getStatus() == status)
+                .forEach(entry -> issueList.add(entry.getValue()));
```
ну и тут какой-нибудь _.collect(Collectors.toList())_

astronautr replied:
Done.

astronautr committed:
Specify each import istead of using .*

Card: #37

astronautr committed:
Change interaction with issueMap

Now, cause we don't need a key .values used, and for more comfortable interaction .collect used.

Card: #37

astronautr committed:
Del unnecessary object initialization and change method return type

Card: #37

kevromster reviewed (approved):
Теперь красота! :)
